Assets/Scripts/HeadBob.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class HeadBob : MonoBehaviour
+{
+    public FpsController fpsController;
+
+    public float bobbingVerticalIntensity = 0.025f;
+    public float bobbingVerticalFrequency = 16f;
+
+    public float bobbingHorizontalIntensity = 0.015f;
+    public float bobbingHorizontalFrequency = 8f;
+
+    public float bobbingSpeed = 9f;
+
+    public float jumpBobIntensity = 0.25f;
+
+    private Vector3 _defaultLocalPosition;
+    private float _timer;
+
+
+    private void Awake()
+    {
+        _defaultLocalPosition = transform.localPosition;
+    }
+
+    private void Update()
+    {
+        var velocity = fpsController.velocity;
+        var planeVelocity = velocity;
+        planeVelocity.y = 0f;
+
+        var planeVelocityMagnitude = planeVelocity.magnitude;
+
+        if (planeVelocityMagnitude > 0f || Mathf.Abs(velocity.y) > 0f)
```

Comment:
Suggest doing some approximate checks here.

```suggestion
        if (!Mathf.Approximately(planeVelocityMagnitude, 0f) || !Mathf.Approximately(Mathf.Abs(velocity.y), 0f))
```

Maybe invert the if statement as well, because there's a lot of negation in that expression